test/unit/mdc-select/mdc-select-enhanced.test.js

```diff
@@ -0,0 +1,654 @@
+
+  MDCSelect.attachTo(fixture);
+  raf.flush();
+
+  const selectedText = fixture.querySelector('.mdc-select__selected-text');
+
+  domEvents.emit(selectedText, 'focus');
+  raf.flush();
+
+  assert.isTrue(fixture.classList.contains(MDCRippleFoundation.cssClasses.BG_FOCUSED));
+  raf.restore();
+});
+
+test('#destroy removes the ripple', function() {
+  if (!supportsCssVariables(window, true)) {
+    this.skip(); // eslint-disable-line no-invalid-this
+    return;
+  }
+
+  const fixture = getFixture();
+  const raf = createMockRaf();
+
+  const component = new MDCSelect(fixture);
+  raf.flush();
+
+  assert.isTrue(fixture.classList.contains(MDCRippleFoundation.cssClasses.ROOT));
+  component.destroy();
+  raf.flush();
+
+  assert.isFalse(fixture.classList.contains(MDCRippleFoundation.cssClasses.ROOT));
+  raf.restore();
+});
+
+test('#destroy cleans up the outline if present', () => {
+  const {component, outline} = setupTest();
+  component.outline_ = outline;
+  component.destroy();
+  td.verify(outline.destroy());
+});
+
+test(`does not instantiate ripple when ${cssClasses.OUTLINED} class is present`, () => {
+  const hasOutline = true;
+  const {component} = setupTest(hasOutline);
+  assert.isUndefined(component.ripple);
+  assert.doesNotThrow(
+    () => component.destroy());
+});
+
+test('adapter#floatLabel adds a class to the label', () => {
+  const {component, label} = setupTest();
+
+  component.getDefaultFoundation().adapter_.floatLabel('foo');
+  td.verify(label.float('foo'));
+});
+
+test('adapter#activateBottomLine adds active class to the bottom line', () => {
+  const {component, bottomLine} = setupTest();
+
+  component.getDefaultFoundation().adapter_.activateBottomLine();
+  td.verify(bottomLine.activate());
+});
+
+test('adapter#deactivateBottomLine removes active class from the bottom line', () => {
+  const {component, bottomLine} = setupTest();
+
+  component.getDefaultFoundation().adapter_.deactivateBottomLine();
+  td.verify(bottomLine.deactivate());
+});
+
+test('adapter#notchOutline proxies labelWidth and isRtl to the outline', () => {
+  const hasOutline = true;
+  const {component, outline} = setupTest(hasOutline);
+  const isRtl = false;
+
+  component.getDefaultFoundation().adapter_.notchOutline(LABEL_WIDTH, isRtl);
+  td.verify(outline.notch(LABEL_WIDTH, isRtl), {times: 1});
+});
+
+test('adapter#notchOutline does not proxy values to the outline if it does not exist', () => {
+  const hasOutline = false;
+  const {component, outline} = setupTest(hasOutline);
+  const isRtl = false;
+
+  component.getDefaultFoundation().adapter_.notchOutline(LABEL_WIDTH, isRtl);
+  td.verify(outline.notch(LABEL_WIDTH, isRtl), {times: 0});
+});
+
+test('adapter#getLabelWidth returns the width of the label', () => {
+  const {component} = setupTest();
+  assert.equal(component.getDefaultFoundation().adapter_.getLabelWidth(), LABEL_WIDTH);
+});
+
+test('adapter#getLabelWidth returns 0 if the label does not exist', () => {
```

Comment:
It looks like we're not actually testing the enhanced-specific adapter methods here, which are probably the most important ones to test in this file in particular... while most of it ends up being covered anyway, the else condition in openMenu and the if condition in closeMenu aren't covered.